dcos/dcos-ui
# DCOS-50715: add manual disabling of add* calculations

## Testing



- Check out the branch
- Add `console.log` calls to the CompositeState add* methods to see in the browser console what happens
- Change export of composite state so you can control it in the browser console:

```diff
- module.exports = new CompositeState({ frameworks: [], slaves: [] });
+ const state = new CompositeState({ frameworks: [], slaves: [] });
+ module.exports = state
+ window.compositeState = state;
```

- Run `window.compositeState.enable()` and `window.compositeState.disable()` in the browser console until you are convinced it works

## Trade-offs



- Choose against EventEmitter, as it seemed harder to implement with an event emitter than with a simple reference counter. 

## Dependencies



## Screenshots



![ezgif com-optimize](https://user-images.githubusercontent.com/1337046/55071455-f3f1df80-5088-11e9-931c-4929979ba6e4.gif)
